### DISCUSSION TITLE
Can't read process.env in GCP cloud run

### DISCUSSION THREAD

**Discussion by YoramGondwe** [vercel/next.js#40619] [Help]
I containerized my application with docker and pushed it to GCP cloud run where its running fine. However, i didn't push the .env.production to my version control and was thinking the nextjs app would pick those variables from the couldrun environmental variables. Unfortunately even i make these environment variables public by adding the key work `NEXT_PUBLIC` it still returns undefined.

my docker file

```
# Get NPM packages
FROM node:lts-alpine3.15 AS dependencies
RUN apk add --no-cache libc6-compat
WORKDIR /app
COPY package.json yarn.lock ./
RUN yarn --frozen-lockfile


# Rebuild the source code only when needed
FROM node:lts-alpine3.15 AS builder
WORKDIR /app
COPY . .
COPY --from=dependencies /app/node_modules ./node_modules
RUN yarn build
#RUN npm run build

# Production image, copy all the files and run next
FROM node:lts-alpine3.15 AS runner
WORKDIR /app


RUN addgroup -g 1001 -S nodejs
RUN adduser -S nextjs -u 1001

COPY --from=builder --chown=nextjs:nodejs /app/.next ./.next
COPY --from=builder /app/node_modules ./node_modules
COPY --from=builder /app/package.json ./package.json
COPY --from=builder /app/public ./public



USER root
EXPOSE 3000


CMD ["yarn", "start"]
```

Where I am trying to access the env in my application.
```
const baseQuery = fetchBaseQuery({
  baseUrl: process.env.NEXT_PUBLIC_BACKEND_API_URL ?? "", prepareHeaders: (headers, { getState }) => {
    const token = (getState() as any).auth.jwtToken
    if (token) {
      headers.set('Authorization', `Bearer ${token}`)
    }
    return headers
  },
})
```

**Comment by Julchu** [comment]
The only way I got it working was by adding them as build arguments (during the `#Rebuild the source code` and before the `RUN yarn build`

> ENV NEXT_PUBLIC_FIREBASE_API_KEY

Also meant I couldn't commit those Dockerfile changes to GitHub, so I'd have to watch out every time
